File: couch/widoki.js
// couchapp push widoki.js http://localhost:14016/marketproduct

var couchapp = require('couchapp');

ddoc = {
    _id: '_design/app'
  , views: {}
  , lists: {}
  , shows: {}
}
module.exports = ddoc;

ddoc.views.popyt = {
  map: function(doc) {
    var val = parseInt(doc.BuyOrders);
    emit([doc.CategoryName, doc.GroupName,doc.Name], val);
  },
  reduce: "_sum"
}

ddoc.views.podaz = {
  map: function(doc) { 
    var val = parseInt(doc.SellOrders);
    emit([doc.CategoryName, doc.GroupName,doc.Name], val);
  },
  reduce: "_sum"
}

File: couch/widoki.py
#!/usr/bin/env python
from couchdb import *
from couchdb.design import ViewDefinition

server = Server('http://sigma.inf.ug.edu.pl:14016')
nazwaBazy = "marketproduct"

db = server[nazwaBazy]

popyt = ViewDefinition(nazwaBazy, 'popyt',
'''function(doc) { 
    var val = parseInt(doc.BuyOrders); 
    emit([doc.CategoryName, doc.GroupName,doc.Name], val); 
  }''' ,"_sum")

podaz = ViewDefinition(nazwaBazy, 'podaz',
'''function(doc) { 
    var val = parseInt(doc.SellOrders); 
    emit([doc.CategoryName, doc.GroupName,doc.Name], val); 
  }''',"_sum")

ViewDefinition.sync_many(db, [popyt,podaz ])
